scripts/migrate-auth.py

```diff
@@ -0,0 +1,192 @@
+"""
+Migration script for Armadillo 2.x -> 3.0
+
+This script copies users and their roles from Fusion Auth to Armadillo. It automatically
+creates users, projects, and permissions. Credentials are requested during execution of
+the script.
+"""
+
+import getopt
+import json
+import sys
+from getpass import getpass
+
+import requests
+from fusionauth.fusionauth_client import FusionAuthClient
+from fusionauth.rest_client import ClientResponse
+from requests import Session
+from simple_term_menu import TerminalMenu
+
+help_string = "python migrate-auth.py --fusion-auth auth_server --armadillo " \
+              "armadillo_server "
+armadillo_url = ""
+auth_url = ""
+
+
+def main(argv):
+    global armadillo_url
+    global auth_url
+
+    try:
+        opts, args = getopt.getopt(argv, "hf:a:", ["fusion-auth=", "armadillo="])
+    except getopt.GetoptError:
+        print(help_string)
+        sys.exit(2)
+
+    for opt, arg in opts:
+        if opt == "-h":
+            print(help_string)
+            sys.exit()
+        elif opt in ("-f", "--fusion-auth"):
+            auth_url = arg
+        elif opt in ("-a", "--armadillo"):
+            armadillo_url = arg.rstrip("/")
+
+    if not auth_url or not armadillo_url:
+        print("All arguments are required: ")
+        print(help_string)
+        sys.exit(2)
+
+    migrate()
+
+
+def migrate():
+    # 1. Set up clients
+    armadillo_client = create_armadillo_client()
+    fusion_client = create_fusion_client()
+
+    # 2. Get production apps that have users
+    users = get_active_registered_users(fusion_client)
+    apps = get_production_applications_with_users(fusion_client, users)
+
+    # 3. Let user pick which app to migrate to this Armadillo server
+    app_to_migrate = choose_application(apps)
```

Comment:
Did you test this with a 'data manager' user? I.e. what permissions must a user have to do this? Data managers have 'SU' in only one or more applications, would they also be able to do this?

Reply:
Oh, and what happens when user doesn't have any applications to choose from?

Reply:
As discussed, we will use API key for now and if it happens that a project's SU wants to do the migration theirselves, then we will add that option later.